### PR TITLE
SAMBRO: color coded cap area

### PR DESCRIPTION
**Pull request by biplovbhandari**
@flavour Please merge...

### CODE REVIEW

**Comment by flavour** [comment]
Just enough here for a last cleanup pre-merge? :)

**Comment by biplovbhandari** [comment]
@flavour Please merge

File: modules/templates/SAMBRO/config.py [hunk 1]
```diff
@@ -262,6 +270,47 @@ def custom_prep(r):
...
+                stable = s3db.gis_style
+                etable = s3db.gis_layer_entity
+                rows = db(etable.instance_type == "gis_layer_feature"). \
+                                                        select(etable.layer_id)
+                query = (stable.layer_id.belongs([row.layer_id for row in rows]))
+                rows = db(query).select(stable.id, stable.style)
+                if rows:
+                    from s3 import IS_JSONS3
+                    for row in rows:
+                        # Start with JSON string (for x-database consistency)
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: Comment no longer valid of course ;)

File: modules/templates/SAMBRO/config.py [hunk 2]
```diff
@@ -262,6 +270,47 @@ def custom_prep(r):
...
+    def customise_cap_warning_priority_resource(r, tablename):
+        
+        s3db = current.s3db
+        def onaccept(form):
+            # Normal onaccept if any
+            form_vars = form.vars
+            color_code = form_vars.color_code
+            if color_code:
+                db = current.db
+                name = form_vars.name
```
Comment (flavour) [inline modules/templates/SAMBRO/config.py]: Can even be moved below the if rows:

**Pull request merged**